controller/userController.js

```diff
@@ -86,7 +92,180 @@ const token = (req, res) => {
     }
 }
 
+const create = (req, res) => {
+    const user = new User(req.body);
+
+    // Check if an user with the provided email exists
+    User.find({ email: user.email }).then(users => {
```

Comment:
User.findOne will return exactly one(if it exists ti returns the first one, email is unique so that is ok) or null otherwise. Use that isntead.

Reply:
Also companyID and admin check are needed.